controlplane/pkg/nsmd/nsmd.go

```diff
@@ -127,6 +127,22 @@ func (nsm *nsmServer) RequestClientConnection(ctx context.Context, request *nsmd
 
 	span.Logger().Infof("Requested client connection to nsmd : %+v", request)
 
+	nsm.Lock()
+	workspace, ok := nsm.workspaces[request.Workspace]
```

Comment:
I think it could be done using Close call on existing workspace to close all used resources and create new ones, it as well will cleanup folder from any existing artifacts.
```suggestion
	// Check if workspace are exists, we should clean resources and reuse-it
	if workspace, ok := nsm.workspaces[request.Workspace]; ok {
		workspace.Close()
	}
```

Reply:
@haiodo @edwarnicke  Do you see any issue with reusing the existing workspace ?  Even in case of normal scenario while creating NSC it looks RequestWorkspace is called 4 times (3 with same socket id and 4th one with different socket id), I think this is because when nsmdp's  Allocate is called from device manager, AllocateRequest contains 4 ContainerRequests for each NSC. so this change would lead to recreating the same workspace again and again. Please let me know.

Reply:
This is unexpected actually, will check.